### PR TITLE
clean up the installation rules

### PR DESCRIPTION
**Pull request by compnerd**
Generalise the install rules and install the missing pieces.  This moves the logic for the installation near the target, and sets up the cross-platform installation rules.  This is dependent on previous PRs and can be rebased once those can be merged.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: Sources/Build/CMakeLists.txt [hunk 1]
```diff
@@ -25,9 +25,11 @@ target_link_libraries(Build PUBLIC
 set_target_properties(Build PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
 
-if(CMAKE_SYSTEM_NAME STREQUAL Windows)
 install(TARGETS Build
```
Comment (aciidb0mb3r) [inline Sources/Build/CMakeLists.txt]: This is not correct. We should only install swift-{build, package, test, run} binaries in the toolchain.

Comment (compnerd) [reply Sources/Build/CMakeLists.txt]: This is under your control: the component system allows you to control that.  By default everything is installed, but, `build-script-impl` can specify the `bins` component to only install the tools.

Comment (aciidb0mb3r) [reply Sources/Build/CMakeLists.txt]: Sure but it's not correct to declare that these libs can be installed in the toolchain. These are purely intermediates.

Comment (compnerd) [reply Sources/Build/CMakeLists.txt]: They are not intermediates when built shared.  So, having the install target with the ability to install them allows both use cases.

**Comment by tomerd** [comment]
@compnerd is this still relevant?

**Comment by compnerd** [comment]
This is still relevant for installation purposes.  I will refresh this in light of the OOB conversation.

**Comment by abertelrud** [comment]
This seems like good generalization — the default is still to build static?  I'm not sure I'm able to decode the CMake in this change...

**Comment by abertelrud** [comment]
@compnerd  Sounds as if you wanted to make some changes to this.  Should this be marked as draft in the meantime?

**Comment by shahmishal** [comment]
We moved our default branch to `main` and deleted `master` branch, GitHub automatically closed the PR. Please update the base branch to `main`.  More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Comment by tomerd** [comment]
Hi @compnerd, the Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please rebase on top of the `main` and we can reopen.

More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Pull request closed without merging**